### PR TITLE
Revises multiple locations describing compound IDs and unique constraints

### LINKED ISSUE #1051: Document composite keys in the find/update/delete sections of the docs
As requested here: https://github.com/prisma/prisma-client-js/issues/488#issuecomment-736690372

### LINKED ISSUE #4588: Clarify docs on @@id and how to use composite IDs in Prisma Client
Per this conversation: https://prisma-company.slack.com/archives/C5Z9TH6N9/p1679587503809509

The docs relating to usage of composite IDs in client code could use a revisit to ensure it is made clear how an `@@id` gets used in client code.

### PR DESCRIPTION
**Pull request by sabinadams**
Closes #4588
Closes #1051

This PR does a few things:

1. Removes the [existing section](https://www.prisma.io/docs/concepts/components/prisma-client/crud#get-record-by-compound-id-or-compound-unique-identifier) describing how to query using a Compound ID and/or Compound Unique Constraint 
2. Adds a clarifying message to the [`@@id` section](https://www.prisma.io/docs/concepts/components/prisma-schema/data-model#defining-an-id-field) about the resulting field's name
3. Adds a clarifying message to the [`@@unique` section](https://www.prisma.io/docs/concepts/components/prisma-schema/data-model#defining-a-unique-field) about the resulting field's name
4. Adds a new page dedicated to using composite IDs and unique constraints in Prisma Client queries

### CODE REVIEW

**Comment by sabinadams** [comment]
@janpio Made a pretty sizeable change to multiple locations relating to `@@id` and `@@unique`. Let me know what you think!

File: content/200-concepts/100-components/01-prisma-schema/04-data-model.mdx [hunk 1]
```diff
@@ -630,6 +630,29 @@ model User {
...
+
+  @@id(name: "fullName", fields: [firstName, lastName])
+}
+```
+
+The `firstName_lastName` field will now be named `fullName` instead.
+
+<Admonition type="info">
+
+Refer to the documentation on [working with composite IDs](/concepts/components/prisma-client/working-with-fields/working-with-composite-ids) to learn how to interact with a composite ID in Prisma Client.
```
Comment (nikolasburk) [inline content/200-concepts/100-components/01-prisma-schema/04-data-model.mdx]: the link here seems to be broken

Comment (nikolasburk) [reply content/200-concepts/100-components/01-prisma-schema/04-data-model.mdx]: I guess it should point to this page: https://docs-git-compound-id-unique-id-section-prisma.vercel.app/concepts/components/prisma-client/working-with-fields/working-with-composite-ids-and-constraints

File: content/200-concepts/100-components/01-prisma-schema/04-data-model.mdx [hunk 2]
```diff
@@ -819,6 +842,32 @@ You can optionally define a [custom unique constraint name](/concepts/components
...
+
+  @@unique(name: "authorTitle", [authorId, title])
+}
+```
+
+The `authorId_title` field will now be named `authorTitle` instead.
+
+<Admonition type="info">
+
+Refer to the documentation on [working with composite unique identifiers](/concepts/components/prisma-client/working-with-fields/working-with-composite-ids) to learn how to interact with a composite unique constraints in Prisma Client.
```
Comment (nikolasburk) [inline content/200-concepts/100-components/01-prisma-schema/04-data-model.mdx]: broken link, probably should link here: https://docs-git-compound-id-unique-id-section-prisma.vercel.app/concepts/components/prisma-client/working-with-fields/working-with-composite-ids-and-constraints

File: content/200-concepts/100-components/02-prisma-client/051-working-with-fields/300-working-with-composite-ids-and-constraints.mdx [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+  postId Int
+  userId Int
+  User   User @relation(fields: [userId], references: [id])
+  Post   Post @relation(fields: [postId], references: [id])
+
+  @@id([postId, userId])
+}
+```
+
+Querying for record from the `Like` table would yield results that look like the following:
```
Comment (nikolasburk) [inline content/200-concepts/100-components/02-prisma-client/051-working-with-fields/300-working-with-composite-ids-and-constraints.mdx]: - "Querying for record" -> bad grammar
- why don't we show a query here? wouldn't that be the most relevant thing to show at this point?

File: content/200-concepts/100-components/02-prisma-client/051-working-with-fields/300-working-with-composite-ids-and-constraints.mdx [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+A composite ID and a composite unique constraint is also usable when creating relational data with `connect` and `connectOrCreate`.
+
+## Filtering records by a compound ID or unique constraint
+
+Although your query results will not display a compound ID or unique constraint as a field, you can use these compound values to filter your queries for unique records:
+
+```ts highlight=3-6;normal
+const like = await prisma.like.findUnique({
+  where: {
+    likeId: {
```
Comment (nikolasburk) [inline content/200-concepts/100-components/02-prisma-client/051-working-with-fields/300-working-with-composite-ids-and-constraints.mdx:94]: I find the connection between this `likeId` and the custom name from before not strong enough. Ideally I'd like to see Prisma Client examples for _both_ scenarios: the one where a custom name is used and one where the default name is used.

**Review by nikolasburk — CHANGES_REQUESTED** [review]

File: content/200-concepts/100-components/01-prisma-schema/04-data-model.mdx [hunk 5]
```diff
@@ -637,6 +637,29 @@ model User {
...
+
+  @@id(name: "fullName", fields: [firstName, lastName])
+}
+```
+
+The `firstName_lastName` field will now be named `fullName` instead.
+
+<Admonition type="info">
+
+Refer to the documentation on [working with composite IDs](/concepts/components/prisma-client/working-with-fields/working-with-composite-ids) to learn how to interact with a composite ID in Prisma Client.
```
Comment (nikolasburk) [inline content/200-concepts/100-components/01-prisma-schema/04-data-model.mdx]:
```suggestion
Refer to the documentation on [working with composite IDs](/concepts/components/prisma-client/working-with-fields/working-with-composite-ids-and-constraints) to learn how to interact with a composite ID in Prisma Client.
```

File: content/200-concepts/100-components/01-prisma-schema/04-data-model.mdx [hunk 6]
```diff
@@ -826,6 +849,32 @@ You can optionally define a [custom unique constraint name](/concepts/components
...
+
+  @@unique(name: "authorTitle", [authorId, title])
+}
+```
+
+The `authorId_title` field will now be named `authorTitle` instead.
+
+<Admonition type="info">
+
+Refer to the documentation on [working with composite unique identifiers](/concepts/components/prisma-client/working-with-fields/working-with-composite-ids) to learn how to interact with a composite unique constraints in Prisma Client.
```
Comment (nikolasburk) [inline content/200-concepts/100-components/01-prisma-schema/04-data-model.mdx]:
```suggestion
Refer to the documentation on [working with composite unique identifiers](/concepts/components/prisma-client/working-with-fields/working-with-composite-ids-and-constraints) to learn how to interact with a composite unique constraints in Prisma Client.
```

File: content/200-concepts/100-components/02-prisma-client/051-working-with-fields/300-working-with-composite-ids-and-constraints.mdx (hunk 3 above)
Comment (nikolasburk) [inline content/200-concepts/100-components/02-prisma-client/051-working-with-fields/300-working-with-composite-ids-and-constraints.mdx]:
```suggestion
Querying for records from the `Like` table would yield results that look like the following:
```

File: content/200-concepts/100-components/02-prisma-client/051-working-with-fields/300-working-with-composite-ids-and-constraints.mdx [hunk 7]
```diff
@@ -0,0 +1,206 @@
...
+  postId Int
+  userId Int
+  User   User @relation(fields: [userId], references: [id])
+  Post   Post @relation(fields: [postId], references: [id])
+
+  @@id([postId, userId])
+}
+```
+
+Querying for records from the `Like` table would yield results that look like the following:
```
Comment (nikolasburk) [inline content/200-concepts/100-components/02-prisma-client/051-working-with-fields/300-working-with-composite-ids-and-constraints.mdx]:
```suggestion
Querying for records from the `Like` table (e.g. using `prisma.like.findMany()`) would return objects that look as follows:
```

**Review by nikolasburk — APPROVED** [review]

**Review by nikolasburk — APPROVED** [review]

**Comment by nikolasburk** [comment]
Thanks @sabinadams 😉 hope you're doing well! 💚

**Pull request merged**